### PR TITLE
#456: Fixed loading double loginVC

### PR DESCRIPTION
**Pull request by ahmed-shehata**
#456: Fixed loading double loginVC

### CODE REVIEW

**Comment by ahmed-shehata** [comment]
👍

File: AtlasUI/AtlasUI/OAuth2AuthorizationHandler.swift [hunk 1]
```diff
@@ -21,9 +21,11 @@ struct OAuth2AuthorizationHandler: AuthorizationHandler {
         let loginViewController = OAuth2LoginViewController(loginURL: self.loginURL, completion: completion)
         let navigationController = UINavigationController(rootViewController: loginViewController)
 
-        Async.main {
-            navigationController.modalPresentationStyle = .OverCurrentContext
-            topViewController.presentViewController(navigationController, animated: true, completion: nil)
+        if !topViewController.isKindOfClass(OAuth2LoginViewController.classForCoder()) {
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/OAuth2AuthorizationHandler.swift]: I would not fix it that way. Still if the user was fast enough, I think he can open it multiple time.

I would use this library to fix these kind of problems
https://github.com/soffes/RateLimit

Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/OAuth2AuthorizationHandler.swift]: Not really, once you call presentViewController, the topViewController would report "LoginViewController". Even if the animation transition did not complete.

Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/OAuth2AuthorizationHandler.swift]: `presentViewController` is in `Async.main` which could give a delay and a chance for the user to retry. 
We have this problem in many places as creating an address. and these problem should be fixed in a proper way not a hacky way :P

Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/OAuth2AuthorizationHandler.swift]: It is not a "hacky" way, but rather checks wherever in the code the "authorize" function was called and then checks to see if the loginVC is already on top. All of this runs in the main thread so there should not be any race condition.

Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/OAuth2AuthorizationHandler.swift]: @bonkey your input :-)

File: AtlasUI/AtlasUI/OAuth2AuthorizationHandler.swift [hunk 2]
```diff
@@ -22,8 +22,10 @@ struct OAuth2AuthorizationHandler: AuthorizationHandler {
         let navigationController = UINavigationController(rootViewController: loginViewController)
 
         Async.main {
-            navigationController.modalPresentationStyle = .OverCurrentContext
-            topViewController.presentViewController(navigationController, animated: true, completion: nil)
+            if !topViewController.isKindOfClass(OAuth2LoginViewController.classForCoder()) {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/OAuth2AuthorizationHandler.swift]: Could be rather `if !(topViewController is OAuth2LoginViewController)`, but it's not the case here. I would just close this PR and talk different approaches.

**Comment by bonkey** [comment]
So, closing, sir.

**Pull request closed without merging**